### PR TITLE
RLCard refactor

### PR DESCRIPTION
**Pull request by WillDudley**
There's some issue here regarding handling of all-zero action masks, several tests raise this whenever `np.random.choice(np.flatnonzero(obs['action_mask'])` is called, or  something like that

### CODE REVIEW

**Comment by jjshoots** [comment]
LGTM.

File: pettingzoo/classic/rlcard_envs/rlcard_base.py [hunk 1]
```diff
@@ -117,7 +123,12 @@ def reset(self, seed=None, return_info=False, options=None):
         self._cumulative_rewards = self._convert_to_dict(
             [0 for _ in range(self.num_agents)]
         )
-        self.dones = self._convert_to_dict([False for _ in range(self.num_agents)])
+        self.terminations = self._convert_to_dict(
+            [False for _ in range(self.num_agents)]
+        )
+        self.truncations = self._convert_to_dict(
+            [False for _ in range(self.num_agents)]
+        )
```
Comment (jjshoots) [inline pettingzoo/classic/rlcard_envs/rlcard_base.py:131]: Found the issue, the environment was not terminating.

**Comment by jjshoots** [comment]
All tests seem to pass now, could you briefly review and lmk what you think? @WillDudley

**Comment by WillDudley** [comment]
All tests pass for me too.

Just have one Q: Are we ok with RLCard envs never truncating? For example, in lucholdem we have
```
    def is_over(self):
        ''' Check if the game is over

        Returns:
            (boolean): True if the game is over
        '''
        alive_players = [1 if p.status=='alive' else 0 for p in self.players]
        # If only one player is alive, the game is over.
        if sum(alive_players) == 1:
            return True

        # If all rounds are finshed
        if self.round_counter >= 2:
            return True
        return False
```

Should we return truncated if self.round_counter >= 2? I'm leaning towards yes but I'm rather unfamiliar with the RLCard envs. I can do this refactor in a few days if we want - got a conf starting now.

Otherwise, LGTM.

**Comment by jjshoots** [comment]
I'm not too familiar with RLCard envs either, but good catch, I think that would be a needed implementation.

**Pull request merged**